clang/include/clang/Sema/Sema.h

```diff
@@ -1357,6 +1363,19 @@ class Sema final {
     // VLAs).
     bool InConditionallyConstantEvaluateContext = false;
 
+    /// Whether we are currently in a context in which temporaries must be
+    /// lifetime-extended (Eg. in a for-range initializer).
+    bool IsInLifetimeExtendingContext = false;
```

Comment:
I think we should be clear here that this is only for unusual lifetime-extension contexts, and not just (for example) in an initializer of a reference.
```suggestion
    /// Whether we are currently in a context in which all temporaries must be
    /// lifetime-extended, even if they're not bound to a reference (for example,
    /// in a for-range initializer).
    bool IsInLifetimeExtendingContext = false;
```